src/lando/main/models/repo.py

```diff
@@ -99,6 +100,28 @@ def raise_if_not(self, repo_scm):
         if repo_scm != self.scm:
             raise self._method_not_supported_for_repo_error
 
+    @property
+    def hg_repo_is_initialized(self):
+        self.raise_if_not(self.HG)
+        try:
+            self.hg._open()
+        except hglib.error.ServerError:
+            logger.info(f"{self} appears to be not initialized.")
+            return False
+        else:
+            return True
+
+    def hg_repo_prepare(self):
```

Comment:
Type hints and docstring.

From reading this it seems like we will likely have a `git_repo_prepare` function and a `git_repo_is_initialized`. Instead, why don't we run a `repo.assert_scm({hg/git})` and then have a more generic `repo_prepare` that does the correct setup based on the SCM for the repo?

Reply:
Git support is out of scope for this PR. Some of the git support was implemented in the prototype worker (which is still in the code base) and will be reimplemented/revisisted/combined with this as part of [bug 1895523](https://bugzilla.mozilla.org/show_bug.cgi?id=1895523). This applies for other shared aspects between git/hg repos.

Reply:
I don't think type hints are necessary for this method. Both self type, and `None` return value are self-evident, and are also not present anywhere else in a similar context.